src/Controller/EnMarche/ElectedRepresentative/DeputyElectedRepresentativeController.php

```diff
@@ -17,8 +17,18 @@ protected function getSpaceType(): string
         return 'deputy';
     }
 
-    protected function getManagedTags(Adherent $adherent): array
+    protected function getManagedZones(Adherent $adherent): array
     {
-        return [$adherent->getManagedDistrict()->getReferentTag()];
+        $managedDistrict = $adherent->getManagedDistrict();
+        if (!$managedDistrict) {
+            return [];
+        }
+
+        $referentTag = $managedDistrict->getReferentTag();
+        if (!$referentTag) {
+            return [];
+        }
+
+        return [$referentTag->getZone()];
```

Comment:
why do we have all these checks? 
we can just return `return [$adherent->getManagedDistrict()->getReferentTag()->getZone()];`, no?